azuredevops/internal/service/build/resource_resource_authorization.go

```diff
@@ -72,41 +78,62 @@ func resourceResourceAuthorizationRead(d *schema.ResourceData, m interface{}) er
 	ctx := context.Background()
 	clients := m.(*client.AggregatedClient)
 
-	authorizedResource, projectID, err := expandAuthorizedResource(d)
+	authorizedResource, projectID, definitionID, err := expandAuthorizedResource(d)
 	if err != nil {
 		return err
 	}
 
-	if !*authorizedResource.Authorized {
+	if definitionID == 0 {
+		if *authorizedResource.Authorized {
+			// (attempt) flatten read result from ado
+			resourceRefs, err := clients.BuildClient.GetProjectResources(ctx, build.GetProjectResourcesArgs{
+				Project: &projectID,
+				Type:    authorizedResource.Type,
+				Id:      authorizedResource.Id,
+			})
+
+			if err != nil {
+				return err
+			}
+
+			// the authorization does no longer exist
+			if len(*resourceRefs) == 0 {
+				d.SetId("")
+				return nil
+			}
+
+			flattenAuthorizedResource(d, &(*resourceRefs)[0], projectID, definitionID)
+			return nil
```

Comment:
I'm friend of sparingly use the `return` statement and have an additional exit point because it's harder to understand a routine if, reading it at the bottom, you're unaware of the possibility that it returned somewhere above. But, and here comes the catch, this heavily depends on the code, because you drive it to the top and always ensure to have only one single exit point at the end of every function, you could end up with code like that 😁 

```c
function()
{
    HRESULT error = S_OK;

    if(SUCCEEDED(Operation1()))
    {
        if(SUCCEEDED(Operation2()))
        {
            if(SUCCEEDED(Operation3()))
            {
                if(SUCCEEDED(Operation4()))
                {
                }
                else
                {
                    error = OPERATION4FAILED;
                }
            }
            else
            {
                error = OPERATION3FAILED;
            }
        }
        else
        {
            error = OPERATION2FAILED;
        }
    }
    else
    {
        error = OPERATION1FAILED;
    }

    return error;
}
```

But here it would make sense to introduce two additional `else` blocks and to remove two `return` statements

![image](https://user-images.githubusercontent.com/14177833/86144362-de722d80-baf5-11ea-99ba-b79d895188e9.png)

Reply:
Good idea :)